Source/WebCore/rendering/RenderReplaced.cpp

```diff
@@ -582,8 +580,8 @@ LayoutUnit RenderReplaced::computeReplacedLogicalWidth(ShouldComputePreferred sh
 
     if (style().logicalWidth().isAuto()) {
         bool computedHeightIsAuto = style().logicalHeight().isAuto();
-        bool hasIntrinsicWidth = constrainedSize.width() > 0;
-        bool hasIntrinsicHeight = constrainedSize.height() > 0;
+        bool hasIntrinsicWidth = constrainedSize.width() > 0 || shouldApplySizeOrInlineSizeContainment();
+        bool hasIntrinsicHeight = constrainedSize.height() > 0 || shouldApplySizeContainment();
```

Comment:
I'm not familiar with this code, and I don't really see the logic behind treating size containment as having a positive intrinsic size. What would be the problem without this?

Reply:
The problem is that if hasIntrinsicWidth is false, the logical width might be calculated by other factors, like ratio * height, etc. That's not what we expected.

Reply:
It would affect some cases, like in contain-intrinsic-size-logical-003.html, 
```
  <canvas class="test cis-none" width="40" height="20"
       data-expected-client-width="0" data-expected-client-height="0"></canvas>
```

Reply:
Yeah, > 0 seems not accurate enough. Maybe we could use `std::optional` to store intrinsic size in the follow-up patch...